File: iceoryx2-ffi/cxx/src/static_config_request_response.cpp
// Copyright (c) 2025 Contributors to the Eclipse Foundation
//
// See the NOTICE file(s) distributed with this work for additional
// information regarding copyright ownership.
//
// This program and the accompanying materials are made available under the
// terms of the Apache Software License 2.0 which is available at
// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
// which is available at https://opensource.org/licenses/MIT.
//
// SPDX-License-Identifier: Apache-2.0 OR MIT

#include "iox2/static_config_request_response.hpp"

namespace iox2 {
auto StaticConfigRequestResponse::request_message_type_details() const -> MessageTypeDetails {
    return MessageTypeDetails(m_value.request_message_type_details);
}

auto StaticConfigRequestResponse::response_message_type_details() const -> MessageTypeDetails {
    return MessageTypeDetails(m_value.response_message_type_details);
}

auto StaticConfigRequestResponse::has_safe_overflow_for_requests() const -> bool {
    return m_value.enable_safe_overflow_for_requests;
}

auto StaticConfigRequestResponse::has_safe_overflow_for_responses() const -> bool {
    return m_value.enable_safe_overflow_for_responses;
}

auto StaticConfigRequestResponse::max_borrowed_responses_per_pending_responses() const -> uint64_t {
    return m_value.max_borrowed_responses_per_pending_response;
}

auto StaticConfigRequestResponse::max_active_requests_per_client() const -> uint64_t {
    return m_value.max_active_requests_per_client;
}

auto StaticConfigRequestResponse::max_response_buffer_size() const -> uint64_t {
    return m_value.max_response_buffer_size;
}

auto StaticConfigRequestResponse::max_loaned_requests() const -> uint64_t {
    return m_value.max_loaned_requests;
}

auto StaticConfigRequestResponse::does_support_fire_and_forget_requests() const -> bool {
    return m_value.enable_fire_and_forget_requests;
}

auto StaticConfigRequestResponse::max_servers() const -> uint64_t {
    return m_value.max_servers;
}

auto StaticConfigRequestResponse::max_clients() const -> uint64_t {
    return m_value.max_clients;
}

auto StaticConfigRequestResponse::max_nodes() const -> uint64_t {
    return m_value.max_nodes;
}

StaticConfigRequestResponse::StaticConfigRequestResponse(iox2_static_config_request_response_t value)
    : m_value { value } {
}
} // namespace iox2

File: iceoryx2-ffi/c/src/api/static_config_request_response.rs
// Copyright (c) 2025 Contributors to the Eclipse Foundation
//
// See the NOTICE file(s) distributed with this work for additional
// information regarding copyright ownership.
//
// This program and the accompanying materials are made available under the
// terms of the Apache Software License 2.0 which is available at
// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
// which is available at https://opensource.org/licenses/MIT.
//
// SPDX-License-Identifier: Apache-2.0 OR MIT

#![allow(non_camel_case_types)]

use iceoryx2::service::static_config::request_response::StaticConfig;

use super::iox2_message_type_details_t;

#[derive(Clone, Copy)]
#[repr(C)]
pub struct iox2_static_config_request_response_t {
    pub enable_safe_overflow_for_requests: bool,
    pub enable_safe_overflow_for_responses: bool,
    pub enable_fire_and_forget_requests: bool,
    pub max_active_requests_per_client: usize,
    pub max_loaned_requests: usize,
    pub max_response_buffer_size: usize,
    pub max_servers: usize,
    pub max_clients: usize,
    pub max_nodes: usize,
    pub max_borrowed_responses_per_pending_response: usize,
    pub request_message_type_details: iox2_message_type_details_t,
    pub response_message_type_details: iox2_message_type_details_t,
}

impl From<&StaticConfig> for iox2_static_config_request_response_t {
    fn from(c: &StaticConfig) -> Self {
        Self {
            enable_safe_overflow_for_requests: c.has_safe_overflow_for_requests(),
            enable_safe_overflow_for_responses: c.has_safe_overflow_for_responses(),
            enable_fire_and_forget_requests: c.does_support_fire_and_forget_requests(),
            max_active_requests_per_client: c.max_active_requests_per_client(),
            max_loaned_requests: c.max_loaned_requests(),
            max_response_buffer_size: c.max_response_buffer_size(),
            max_servers: c.max_servers(),
            max_clients: c.max_clients(),
            max_nodes: c.max_nodes(),
            max_borrowed_responses_per_pending_response: c
                .max_borrowed_responses_per_pending_response(),
            request_message_type_details: c.request_message_type_details().into(),
            response_message_type_details: c.response_message_type_details().into(),
        }
    }
}
